File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnRole.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.verbnet.sql;

/**
 * VerbNet role
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class VnRole
{
	/**
	 * Role type
	 */
	public final String roleType;

	/**
	 * Role selectional restrictions
	 */
	public final String selectionRestrictions;

	/**
	 * Constructor
	 *
	 * @param roleType              role type
	 * @param selectionRestrictions selectional restriction (XML)
	 */
	VnRole(final String roleType, final String selectionRestrictions)
	{
		this.roleType = roleType;
		this.selectionRestrictions = selectionRestrictions;
	}
}

File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnRole.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.verbnet.sql

/**
 * VerbNet role
 *
 * @param roleType              role type
 * @param selectionRestrictions selectional restriction (XML)
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
class VnRole(
    val roleType: String,
    val selectionRestrictions: String?
) 